src/afl-fuzz-stats.c

```diff
@@ -543,74 +562,85 @@ void show_stats(afl_state_t *afl) {
+             stringify_int(IB(5), afl->stage_cycles[STAGE_ARITH32]));
 
   SAYF(bV bSTOP " arithmetics : " cRST "%-36s " bSTG bV bSTOP
                 "  pend fav : " cRST "%-10s" bSTG       bV "\n",
-       tmp, DI(afl->pending_favored));
+       tmp, stringify_int(IB(0), afl->pending_favored));
 
   if (!afl->skip_deterministic)
-    sprintf(tmp, "%s/%s, %s/%s, %s/%s", DI(afl->stage_finds[STAGE_INTEREST8]),
-            DI(afl->stage_cycles[STAGE_INTEREST8]),
-            DI(afl->stage_finds[STAGE_INTEREST16]),
-            DI(afl->stage_cycles[STAGE_INTEREST16]),
-            DI(afl->stage_finds[STAGE_INTEREST32]),
-            DI(afl->stage_cycles[STAGE_INTEREST32]));
+    sprintf(tmp, "%s/%s, %s/%s, %s/%s",
+            stringify_int(IB(0), afl->stage_finds[STAGE_INTEREST8]),
+            stringify_int(IB(1), afl->stage_cycles[STAGE_INTEREST8]),
+            stringify_int(IB(2), afl->stage_finds[STAGE_INTEREST16]),
+            stringify_int(IB(3), afl->stage_cycles[STAGE_INTEREST16]),
+            stringify_int(IB(4), afl->stage_finds[STAGE_INTEREST32]),
+            stringify_int(IB(5), afl->stage_cycles[STAGE_INTEREST32]));
 
   SAYF(bV bSTOP "  known ints : " cRST "%-36s " bSTG bV bSTOP
                 " own finds : " cRST "%-10s" bSTG       bV "\n",
-       tmp, DI(afl->queued_discovered));
+       tmp, stringify_int(IB(0), afl->queued_discovered));
 
   if (!afl->skip_deterministic)
-    sprintf(tmp, "%s/%s, %s/%s, %s/%s", DI(afl->stage_finds[STAGE_EXTRAS_UO]),
-            DI(afl->stage_cycles[STAGE_EXTRAS_UO]),
-            DI(afl->stage_finds[STAGE_EXTRAS_UI]),
-            DI(afl->stage_cycles[STAGE_EXTRAS_UI]),
-            DI(afl->stage_finds[STAGE_EXTRAS_AO]),
-            DI(afl->stage_cycles[STAGE_EXTRAS_AO]));
+    snprintf(tmp, sizeof(tmp), "%s/%s, %s/%s, %s/%s",
+             stringify_int(IB(0), afl->stage_finds[STAGE_EXTRAS_UO]),
+             stringify_int(IB(1), afl->stage_cycles[STAGE_EXTRAS_UO]),
+             stringify_int(IB(2), afl->stage_finds[STAGE_EXTRAS_UI]),
+             stringify_int(IB(3), afl->stage_cycles[STAGE_EXTRAS_UI]),
+             stringify_int(IB(4), afl->stage_finds[STAGE_EXTRAS_AO]),
+             stringify_int(IB(5), afl->stage_cycles[STAGE_EXTRAS_AO]));
 
   SAYF(bV bSTOP "  dictionary : " cRST "%-36s " bSTG bV bSTOP
                 "  imported : " cRST "%-10s" bSTG       bV "\n",
-       tmp, afl->sync_id ? DI(afl->queued_imported) : (u8 *)"n/a");
+       tmp,
+       afl->sync_id ? stringify_int(IB(0), afl->queued_imported) : (u8 *)"n/a");
 
-  sprintf(
-      tmp, "%s/%s, %s/%s, %s/%s", DI(afl->stage_finds[STAGE_HAVOC]),
-      DI(afl->stage_cycles[STAGE_HAVOC]), DI(afl->stage_finds[STAGE_SPLICE]),
-      DI(afl->stage_cycles[STAGE_SPLICE]), DI(afl->stage_finds[STAGE_RADAMSA]),
-      DI(afl->stage_cycles[STAGE_RADAMSA]));
+  snprintf(tmp, sizeof(tmp), "%s/%s, %s/%s, %s/%s",
+           stringify_int(IB(0), afl->stage_finds[STAGE_HAVOC]),
```

Comment:
since we always use bytes of length 16, we could change the function to an unsafe version expecting 16 bytes. This would be quicker.

Reply:
same here

Reply:
ok.